ome_zarr.py

```diff
@@ -44,71 +49,112 @@ def napari_get_reader(path: PathLike) -> Optional[ReaderFunction]:
-        metadata['name'] = [ch['label'] for ch in image_data['channels']]
-        metadata['visible'] = [ch['active'] for ch in image_data['channels']]
-    except Exception:
-        pass
-
-    return metadata
-
-
-def load_omero_zarr(path):
-    zarr_path = path.endswith("/") and path or f"{path}/"
-    attrs_path = zarr_path + ".zattrs"
-    root_attrs = requests.get(attrs_path).json()
-
-    resolutions = ["0"]  # TODO: could be first alphanumeric dataset on err
-    try:
-        print('root_attrs', root_attrs)
-        if 'multiscales' in root_attrs:
-            datasets = root_attrs['multiscales'][0]['datasets']
-            resolutions = [d['path'] for d in datasets]
-        print('resolutions', resolutions)
-    except Exception as e:
-        raise e
-
-    pyramid = []
-    for resolution in resolutions:
-        # data.shape is (t, c, z, y, x) by convention
-        data = da.from_zarr(f"{zarr_path}{resolution}")
-        chunk_sizes = [str(c[0]) + (" (+ %s)" % c[-1] if c[-1] != c[0] else '') for c in data.chunks]
-        print('resolution', resolution, 'shape (t, c, z, y, x)', data.shape, 'chunks', chunk_sizes, 'dtype', data.dtype)
-        pyramid.append(data)
-
-    metadata = load_omero_metadata(zarr_path)
-
-    return(pyramid, {'channel_axis': 1, **metadata})
-
-    # TODO: would be nice to be able to:
-    # viewer.dims.set_axis_label(0, 'T')
-    # viewer.dims.set_point(0, image_data['rdefs']['defaultT'])
-    # viewer.dims.set_axis_label(1, 'Z')
-    # viewer.dims.set_point(1, image_data['rdefs']['defaultZ'])
+    result = urlparse(path)
+    if result.scheme in ("", "file"):
+        # Strips 'file://' if necessary
+        instance = LocalZarr(result.path)
+    else:
+        instance = RemoteZarr(path)
+
+    if instance.is_zarr():
+        return instance.get_reader_function()
+
+
+class BaseZarr:
+
+    def __init__(self, path):
+        self.zarr_path = path.endswith("/") and path or f"{path}/"
+        self.zarray = self.get_json(".zarray")
+        self.zgroup = self.get_json(".zgroup")
+        if self.is_zarr():
+            self.image_data = self.get_json("omero.json")
+            self.root_attrs = self.get_json(".zattrs")
+
+    def is_zarr(self):
+        return self.zarray or self.zgroup
+
+    def get_json(self, subpath):
+        raise NotImplementedError("unknown")
+
+    def get_reader_function(self):
+        if not self.is_zarr():
+            raise Exception("not a zarr")
+        return self.reader_function
+
+    def reader_function(self, path: PathLike) -> List[LayerData]:
+        """Take a path or list of paths and return a list of LayerData tuples."""
+        if isinstance(path, list):
+            path = path[0]
+        return [self.load_omero_zarr()]
+
+    def load_omero_metadata(self):
+        """Load OMERO metadata as json and convert for napari"""
+        metadata = {}
```

Comment:
```suggestion
        metadata = {}
        image_data = self.image_data
```

Reply:
This will fix travis

Reply:
Went ahead and inlined `channels` as well.